js/content/community.js

```
@@ -3459,6 +3459,128 @@ class WorkshopPageClass {
+                        </button>
+                    </div>
+                </div>
+            </div>`;
+
+        Array.from(adminTools).forEach(node => {
+            node = node.parentNode.parentNode.parentNode.parentNode;
+            if (node.querySelector("#es_del_announcements")) { return };
+            node = node.querySelector(".rightbox_content_header");
+
+            HTML.afterEnd(node, delAnnouncementsBtns);
+            node.parentNode.querySelector("[id=es_del_announcements]").addEventListener("click", () => deleteAnnouncements(false), false);
+            node.parentNode.querySelector("[id=es_del_hidden]").addEventListener("click", () => deleteAnnouncements(true), false);
+        });
+
+        function deleteAnnouncements(hidden = false) {
+            let i = 0;
+            let total = 0;
+            let modalTitle = hidden ? Localization.str.del_hidden_announcements : Localization.str.del_announcements;
+
+            ExtensionLayer.runInPageContext(`function(){
+                let prompt = ShowConfirmDialog("${modalTitle}", \`${Localization.str.del_announcements_confirm}\`);
+                prompt.done(function(result) {
+                    if (result == "OK") {
+                        Messenger.postMessage("delEmAll");
+                    }
+                });
+            }`);
+
+            function updateWaitDialog() {
+                ExtensionLayer.runInPageContext(`function() {
+                    if (window.dialog) {
+                        window.dialog.Dismiss();
+                    }
+
+                    window.dialog = ShowBlockingWaitDialog("${modalTitle}", \`${Localization.str.del_announcements_loading.replace("__i__", i).replace("__total__", total)}\`);
+                }`)
+            }
+
+            function deleteAnnouncement(url) {
+                return new Promise(async resolve => {
+                    await RequestData.getHttp(url);
```

Why the new promise? and what happens if getHttp rejects?

---

uncaught promise rejection

---

From the user's perspective, they don't see the console red text. So this would be "failing silently".